### DISCUSSION TITLE
Bulk poster generation

### DISCUSSION THREAD

**Discussion by apecollector** [google/model-viewer#3381] [Q&A]
I have several thousand models that I'd like to generate poster images for. Is there any documented or easy way to do so without manually opening them up in the editor?

**Comment by mitdave95** [comment]
Run MV inside a headless chrome and using js use the below flow
1. Update the model
2. Listen for model loaded event
3. Create poster
4. Save it on your local or server and wait for callback
Repeat step 1

**Reply by elalish** [reply]
For a code example, have a look at how we do it for our Twitter card generator: https://github.com/google/model-viewer/blob/master/packages/modelviewer.dev/examples/twitter/generator.js#L94

**Reply by apecollector** [reply]
Thanks @mitdave95 and @elalish! I'll look into this, hopefully I can contribute something back either in terms of documentation or actual code to accomplish bulk poster generation.

**Reply by apecollector** [reply]
@mitdave95 and @elalish I got it working using puppeteer, but only with `headless: false` and running on my laptop, I see the same is being done in the fidelity tests: https://github.com/google/model-viewer/blob/master/packages/render-fidelity-tools/src/artifact-creator.ts#L268-L275

Ideally I'd be able to use a true headless chrome so it would work on our servers where the thousands of files are actually stored in order to save bandwidth.

**Reply by elalish** [reply]
That's strange that headless mode doesn't work. I think we only changed that for Filament originally. Of course the server probably doesn't have a GPU, so you'll be stuck with SwiftShader, which is anything but swift and may have more bugs. What error do you get with headless?

**Reply by apecollector** [reply]
Actually I just tried it again and it worked! I think it was actually that I needed to add this so the downloads worked:

```
  await page._client.send("Page.setDownloadBehavior", {
    behavior: "allow",
    downloadPath: "./images",
  });
```

**Reply by madjin** [reply]
I'm working on the exact same kind of task and haven't quite figured it out. I think bulk poster generation would be useful for many types of applications, going to keep going off the breadcrumbs here and chatGPT to figure it out while hoping for better docs / code around this use case someday

**Comment by lalalune** [comment]
https://github.com/Shopify/screenshot-glb
Shopify has this going in a way that you can script with shell
